### ISSUE TITLE
Paranoid join tables in through relationships do not work with with_deleted

### ISSUE THREAD

**Issue by pauldruziak** [ActsAsParanoid/acts_as_paranoid#337]
I added tests here to confirm it, but it failed in all versions: https://github.com/ActsAsParanoid/acts_as_paranoid/pull/336

So, then I wrote this test in my project and it failed in 7.0 and passed in 6.1:
```ruby
# frozen_string_literal: true

require "spec_helper"

describe "Paranoid" do
  class ParanoidManyManyParentLeft < ActiveRecord::Base
    has_many :paranoid_many_many_children, inverse_of: :paranoid_many_many_parent_left
    has_many :paranoid_many_many_parent_rights, through: :paranoid_many_many_children
  end

  class ParanoidManyManyParentRight < ActiveRecord::Base
    acts_as_paranoid
    has_many :paranoid_many_many_children, dependent: :destroy, inverse_of: :paranoid_many_many_parent_right
    has_many :paranoid_many_many_parent_lefts, through: :paranoid_many_many_children
  end

  class ParanoidManyManyChild < ActiveRecord::Base
    acts_as_paranoid
    belongs_to :paranoid_many_many_parent_left, inverse_of: :paranoid_many_many_children
    belongs_to :paranoid_many_many_parent_right, inverse_of: :paranoid_many_many_children
  end

  before do
    ActiveRecord::Schema.define(version: 1) do
      create_table :paranoid_many_many_parent_lefts do |t|
        t.string :name
      end

      create_table :paranoid_many_many_parent_rights do |t|
        t.string :name
        t.datetime :deleted_at
      end

      create_table :paranoid_many_many_children do |t|
        t.integer :paranoid_many_many_parent_left_id
        t.integer :paranoid_many_many_parent_right_id
        t.datetime :deleted_at
      end
    end
  end

  it "should work with has_many through" do
    paranoid_left = ParanoidManyManyParentLeft.create!
    paranoid_right = paranoid_left.paranoid_many_many_parent_rights.create!
    paranoid_right.destroy

    expect(paranoid_left.paranoid_many_many_parent_rights.only_deleted).to eq([paranoid_right])
  end
end
```

The only difference that I see right now is that `acts_as_paraniod` uses `SQLite` for tests, while my project uses `PostgreSQL`. I'm going to continue the investigation. And next step I see is creating a new Rails project with `SQLite` to see how this will behave there.

**Comment by mvz** [comment]
Hi @pauldruziak I tried the test you posted in this issue and can confirm that it fails with Rails 6.1, 7.0 and 7.1.

Which version of `acts_as_paranoid` are you using in your project?

**Comment by mvz** [comment]
@pauldruziak then it makes sense that your tests in #336 failed: The current release is 0.10.0.

Can you confirm that version 0.10.0 also does the wrong thing, both for Rails 6.1 and 7.0?

**Comment by mvz** [comment]
Thanks for confirming that @pauldruziak.

**Comment by mvz** [comment]
@pauldruziak this should be fixed in version 0.10.1 which I just released.
